src/lib.rs

```diff
@@ -285,30 +316,85 @@ pub fn encode_config_buf<T: ?Sized + AsRef<[u8]>>(input: &T, config: Config, buf
     buf_bytes.resize(orig_buf_len.checked_add(encoded_size)
                          .expect("usize overflow when calculating expanded buffer size"), 0);
 
-    let b64_output = &mut buf_bytes[orig_buf_len..];
+    let mut b64_output = &mut buf_bytes[orig_buf_len..];
 
-    let encoded_bytes = encode_with_padding(input_bytes, b64_output, config.char_set.encode_table(),
-                                            config.pad);
+    encode_with_padding_line_wrap(&input_bytes, &config, encoded_size, &mut b64_output);
+}
 
-    if let LineWrap::Wrap(line_len, line_end) = config.line_wrap {
-        line_wrap(b64_output, encoded_bytes, line_len, line_end);
-    }
+/// Encode arbitrary octets as base64.
+/// Writes into the supplied output buffer.
+///
+/// This is useful if you wish to avoid allocation entirely (e.g. encoding into a stack-resident
+/// or statically-allocated buffer).
+///
+/// # Panics
+///
+/// If `output` is too small to hold the encoded version of `input`, a panic will result.
+///
+/// # Example
+/// 
+/// ```rust
+/// extern crate base64;
+/// 
+/// fn main() {
+///     let s = b"hello internet!";
+///     let mut buf = Vec::new();
+///     // make sure we'll have a slice big enough for base64 + padding
+///     buf.resize(s.len() * 4 / 3 + 4, 0);
```

Comment:
Oh I like this definition in the docs! But would it be worth just doing `Vec::with_capacity(...)` instead of the resize dance?

Reply:
`with_capacity` specifies the internal capacity but doesn't extend the length of the vec. I need the elements to really be there so that when I take a mutable slice it includes all those bytes (set to 0 here).